### PR TITLE
Update AppEngine Samples

### PR DESCRIPTION
**Pull request by frankyn**
Updating the following app.yaml, Gemfile.lock, and outdated files for App Engine Samples:

- [x]  Analytics Sample
- [x]  Cloud SQL
- [x]  Datastore
- [x]  Endpoints
- [x]  Hello World
- [x]  Memcached - ignored
- [x]  Mailgun
- [x]  Metadata Server
- [x]  PubSub
- [x]  SendGrid
- [x]  Static Files (Rails)
- [x]  Static Files (Sinatra)
- [x]  Storage
- [x]  Twilio

### CODE REVIEW

**Comment by frankyn** [comment]
Updated App Engine samples

**PTAL** @remi

File: appengine/sendgrid/app.rb [hunk 1]
```diff
@@ -31,19 +31,25 @@
...
+  subject = "Hello from Google Cloud Ruby SendGrid Sample"
+  content = SendGrid::Content.new type:  "text/plain",
+                                  value: "Congratulations it works!"
+
+  # Define the new email with provided information
+  mail = SendGrid::Mail.new(from, subject, to, content)
 
-  sendgrid = SendGrid::Client.new api_key: SENDGRID_API_KEY
+  # Create a new API Client to send the new email
+  sg = SendGrid::API.new api_key: SENDGRID_API_KEY
```
Comment (remi) [inline appengine/sendgrid/app.rb]: Can we rename this back to `sendgrid` instead of abbreviating?

File: appengine/sendgrid/app.yaml [hunk 2]
```diff
@@ -13,8 +13,8 @@
 # limitations under the License.
 
 runtime: ruby
-vm: true
-entrypoint: bundle exec ruby app.rb -p 8080
+env: flex
+entrypoint: bundle exec ruby app.rb -p $PORT
```
Comment (remi) [inline appengine/sendgrid/app.yaml]: :+1:

**Comment by frankyn** [comment]
PTAL @remi

**Comment by remi** [comment]
👍  LGTM

**Comment by remi** [comment]
**reminder:** _do not merge_ 😄

**Pull request merged**